src/neuroconv/tools/nwb_helpers/_models/_hdf5_dataset_models.py

```diff
@@ -0,0 +1,76 @@
+"""Base Pydantic models for the HDF5DatasetConfiguration."""
+from typing import Any, Dict, Literal, Union
+
+import h5py
+from nwbinspector.utils import is_module_installed
+from pydantic import Field
+
+from ._base_dataset_models import DatasetConfiguration
+
+_base_hdf5_filters = set(h5py.filters.decode) - set(
+    (
+        "shuffle",  # controlled via H5DataIO
+        "fletcher32",  # controlled via H5DataIO
+        "scaleoffset",  # enforced indrectly by HDMF/PyNWB data types
+    )
+)
+_available_hdf5_filters = set(_base_hdf5_filters)
+if is_module_installed(module_name="hdf5plugin"):
+    import hdf5plugin
+
+    _available_hdf5_filters = _available_hdf5_filters | set(
+        (str(hdf5plugin_filter).rstrip("'>").split(".")[-1] for hdf5plugin_filter in hdf5plugin.get_filters())
```

Comment:
Depends on the users system. You could define or dynamically import any arbitrary number of them, but they do all get registered along this pathway, which is how the plugin library detects them

Reply:
Some are also only available on certain platforms/versions/architectures

Reply:
I mean the mapping between  what is available  in `hdf5plugin.get_filters()` to the variable  `_available_hdf5_filters`. I am proposing just having a translation dict instead of the parsing you have here. It is easier to change and update if there are changes in what you get in `hdf5plugin.get_filters()` or if there is an error. Though, I think this work if the number of elements is <15 and after that it becomes too much.

Reply:
This is a design of how it ought to look in the GUIDE as a dropdown selection

My results from `_available_hdf5_filter`:

```python
{'BZip2',
 'Bitshuffle',
 'Blosc',
 'Blosc2',
 'FciDecomp',
 'LZ4',
 'SZ',
 'SZ3',
 'Zfp',
 'Zstd',
 'gzip',
 'lzf'}
```

Without this nicer parsing it looks like this

```
["<class 'hdf5plugin._filters.Bitshuffle'>",
 "<class 'hdf5plugin._filters.Blosc'>",
 "<class 'hdf5plugin._filters.Blosc2'>",
 "<class 'hdf5plugin._filters.BZip2'>",
 "<class 'hdf5plugin._filters.FciDecomp'>",
 "<class 'hdf5plugin._filters.LZ4'>",
 "<class 'hdf5plugin._filters.SZ'>",
 "<class 'hdf5plugin._filters.SZ3'>",
 "<class 'hdf5plugin._filters.Zfp'>",
 "<class 'hdf5plugin._filters.Zstd'>"]
```

Oh also, the base filters do NOT have that same class form since they are built-in to `h5py`, you just refer to them with the 'parsed' string form such as `"gzip"`

Reply:
Or are you simply suggesting to turn the entier thing (the zarr one too) into a dict where the keys are the parsed versions and the values are the actual classes?

Reply:
Sure, will do